corehq/apps/callcenter/views.py

```diff
@@ -0,0 +1,89 @@
+from django.utils.translation import ugettext
+
+
+from corehq.apps.groups.models import Group
+from corehq.apps.locations.models import SQLLocation, Location
+from corehq.apps.reports.filters.api import EmwfOptionsView
+from corehq.apps.reports.util import _report_user_dict
+from corehq.apps.users.cases import get_wrapped_owner
+from corehq.apps.users.models import CommCareUser
+from corehq.toggles import CALL_CENTER_LOCATION_OWNERS
+from dimagi.utils.decorators.memoized import memoized
+
+
+class _CallCenterOwnerOptionsUtils(object):
+    def __init__(self, domain):
+        self.domain = domain
+
+    def user_tuple(self, user):
+        user = _report_user_dict(user)
+        name = "%s [user]" % user['username_in_report']
+        return (user['user_id'], name)
+
+    def reporting_group_tuple(self, group):
+        return (group['_id'], '%s [group]' % group['name'])
+
+    def location_tuple(self, location):
+        return self.reporting_group_tuple(location.case_sharing_group_object())
+
+    @property
+    @memoized
+    def static_options(self):
+        from corehq.apps.domain.forms import USE_LOCATION_CHOICE
+        from corehq.apps.domain.forms import USE_PARENT_LOCATION_CHOICE
+
+        if CALL_CENTER_LOCATION_OWNERS.enabled(self.domain):
+            return [
+                (USE_LOCATION_CHOICE, ugettext("user's location [location]")),
+                (USE_PARENT_LOCATION_CHOICE, ugettext("user's location's parent [location]")),
+            ]
+        return []
+
+
+class CallCenterOwnerOptionsView(EmwfOptionsView):
+    url_name = "call_center_owner_options"
+
+    @property
+    @memoized
+    def utils(self):
+        return _CallCenterOwnerOptionsUtils(self.domain)
+
+    def get_locations_query(self, query):
+        return SQLLocation.objects.filter_path_by_user_input(self.domain, query).filter(
```

Comment:
Probably not important. I don't know if you are ever going to want this to be `location_safe`, but if you do, you'll want to also add `accessible_to_user` to this.